src/android/android-uvc.cpp

```diff
@@ -0,0 +1,598 @@
+                for(auto&& dev : usb_host::android_device_watcher::get_device_list())
+                {
+                    std::shared_ptr<usbhost_uvc_device> uvc(new usbhost_uvc_device(),
+                                                            [](usbhost_uvc_device *ptr) {usbhost_close(ptr); });
+                    uvc->device = dev;
+                    uvc->vid = dev->get_vid();
+                    uvc->pid = dev->get_pid();
+                    uvc_devices.push_back(uvc);
+                }
+
+                for (auto device : uvc_devices) {
+
+                    usbhost_open(device.get(), mi);
+
+                    if (device->deviceData.ctrl_if.bInterfaceNumber == mi) {
+                        _device = device;
+
+                        for (auto ct = _device->deviceData.ctrl_if.input_term_descs;
+                             ct; ct = ct->next) {
+                            _input_terminal = ct->bTerminalID;
+                        }
+
+                        for (auto pu = _device->deviceData.ctrl_if.processing_unit_descs;
+                             pu; pu = pu->next) {
+                            _processing_unit = pu->bUnitID;
+                        }
+
+                        for (auto eu = _device->deviceData.ctrl_if.extension_unit_descs;
+                             eu; eu = eu->next) {
+                            _extension_unit = eu->bUnitID;
+                        }
+
+                        _device->device->claim_interface(mi);
+
+                        _power_state = D0;
+                        return;
+                    }
+                }
+
+                throw std::runtime_error("Device not found!");
+            }
+            if (state == D3 && _power_state == D0) {
+                _device.reset();
+                _power_state = D3;
+            }
+        }
+
+        std::vector<stream_profile> android_uvc_device::get_profiles() const {
+            std::vector<stream_profile> results;
+            int i = 0;
+
+            check_connection();
+
+            if (get_power_state() != D0) {
+                throw std::runtime_error("Device must be powered to query supported profiles!");
+            }
+
+            // Return linked list of uvc_format_t of all available formats inside device
+            uvc_format_t *formats = NULL;
+            usbhost_get_available_formats_all(_device.get(), &formats);
+
+            uvc_format_t *curFormat = formats;
+            while (curFormat != NULL) {
+                stream_profile sp;
+                sp.width = curFormat->width;
+                sp.height = curFormat->height;
+                sp.fps = curFormat->fps;
+                sp.format = curFormat->fourcc;
+                results.push_back(sp);
+                curFormat = curFormat->next;
+            }
+
+            usbhost_free_formats(formats);
+
+            return results;
+        }
+
+
+        //TODO: fix this
```

Comment:
Do we know what this `TODO` referred to? I'm not sure what needs fixing here

Reply:
No clue. Removed in 310179b5b